lib/msf/core/handler/reverse_tcp.rb

```diff
@@ -99,7 +104,8 @@ def payload_uri
   def listener_uri(addr = datastore['ReverseListenerBindAddress'])
     addr = datastore['LHOST'] if addr.nil? || addr.empty?
     uri_host = Rex::Socket.is_ipv6?(addr) ? "[#{addr}]" : addr
-    "tcp://#{uri_host}:#{bind_port}"
+    via = via_string(listener_sock.client) if listener_sock.respond_to?(:client)
+    "tcp://#{uri_host}:#{bind_port} #{via}"
```

Comment:
Going to have a trailing space here if `via` is still `nil` after all these checks, might want to strip that out.